File: algorithm/001-two_sum/kotlin/solution.kt
class Solution {
  fun twoSum(nums: IntArray, target: Int): IntArray {
    val complements = HashMap<Int, Int>()
    for (i in nums.indices) {
      var c : Int = target - nums[i]
      if (complements.containsKey(c)) {
        return intArrayOf(complements[c]!!, i)
      } else {
        complements.put(nums[i], i)
      }
    }

    return intArrayOf()
  }
}

File: algorithm/001-two_sum/swift/solution.swift
class Solution {
  func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
    var complements = [Int: Int]()
    for i in 0...(nums.count - 1) {
      let c = target - nums[i]
      if (complements[c] != nil) {
        return [complements[c]!, i]
      } else {
        complements[nums[i]] = i
      }
    }
    return []
  }
}